client/src/main/java/se/fortnox/reactivewizard/client/HttpClient.java

```diff
@@ -169,9 +177,35 @@ public Object invoke(Object proxy, Method method, Object[] arguments) {
         publisher = withRetry(request, flux).onErrorResume(e -> convertError(request, e));
 
         if (Single.class.isAssignableFrom(method.getReturnType())) {
-            return RxReactiveStreams.toSingle(publisher);
+            return new SingleWithResponse(RxReactiveStreams.toSingle(publisher), rawResponse);
+        }
+        return new ObservableWithResponse(RxReactiveStreams.toObservable(publisher), rawResponse);
+    }
+
+    /** Should be used with
+     * @param source the source observable, must be observable returned from api call
+     * @param <T> the type of data that should be returned in the call
+     * @return an observable that along with the data passes the response object from netty
+     */
+    public static <T> Observable<Response<T>> fullResponse(Observable<T> source) {
```

Comment:
when are these used?

Reply:
Only in tests.
Should be used in systems that needs response metadata along with the actual data